mountpoint-s3-client/src/mock_client.rs

```diff
@@ -99,6 +99,11 @@ impl MockClient {
     pub fn is_upload_in_progress(&self, key: &str) -> bool {
         self.in_progress_uploads.read().unwrap().contains(key)
     }
+
+    /// Returns the objects storage class
+    pub fn get_object_storage_class(&self, key: &str) -> Option<String> {
```

Comment:
This should probably return an error if the key is not found, rather than panicking.